### PR TITLE
Add a flag to include stop string in output text

### LINKED ISSUE #1973: Flag to include stop string in output
Currently vLLM cuts out the stop string from output text (https://github.com/vllm-project/vllm/blob/main/vllm/engine/llm_engine.py#L683-L689). Reason for this add:
1. In case there are multiple stop strings, we want to understand which stop string caused the stop of generation
2. This also matches the behavior of TGI

### PR DESCRIPTION
**Pull request by yunfeng-scale**
Fixes https://github.com/vllm-project/vllm/issues/1973

tested by running server locally

### CODE REVIEW

**Comment by simon-mo** [comment]
this looks good! what's the variable name in TGI? How about placing it near the `stop` parameters and name it like `include_stop_in_output` or others you would like.

**Comment by yunfeng-scale** [comment]
> this looks good! what's the variable name in TGI? How about placing it near the `stop` parameters and name it like `include_stop_in_output` or others you would like.

thanks for the review. TGI does not have this variable. it's default behavior is to include stop string in output.

**Comment by yunfeng-scale** [comment]
@WoosukKwon @zhuohan123 would you take a look?

File: vllm/sampling_params.py [hunk 1]
```diff
@@ -70,6 +71,8 @@ class SamplingParams:
         stop_token_ids: List of tokens that stop the generation when they are
             generated. The returned output will contain the stop tokens unless
             the stop tokens are special tokens.
+        include_stop_in_output: Whether to include the stop strings in output
```
Comment (WoosukKwon) [inline vllm/sampling_params.py:74]: `include_stop_str_in_output` looks a bit clearer, WDYT?

```suggestion
        include_stop_str_in_output: Whether to include the stop strings in output
```

**Review by WoosukKwon — COMMENTED** [review]
Hi @yunfeng-scale, thanks for submitting the PR! I think this makes sense. A friend of mine actually requested this 😂 

Could you resolve the merge conflict and see my comment on the variable name? Thanks.

**Review by WoosukKwon — APPROVED** [review]
LGTM! Thanks for the contribution!

**Pull request merged**